### DISCUSSION TITLE
X axis goes too far when leveling, hits my cable chain

### DISCUSSION THREAD

**Discussion by TekoXVI** [Jyers/Marlin#1883] [Q&A] [answered]
I have a cable chain mount on the hot end that hits the side when the hotend moves to the right while leveling (while it's trying to get the probe to the very right side of the bed). How can I fix it so it doesn't try to move so far to the right? This is on an Ender 3 V2 with CR touch, 15x15 HS UBL firmware.

**Comment by BlueWings172** [comment, marked as answer]
@TekoXVI 

Had similar issues few months ago. You need to change X_Max_POS value in firmware then compile it. Check [this thread](https://github.com/Jyers/Marlin/discussions/1716) of some details.

If you have not compiled firmware before, here are some instructions in[ this post](https://github.com/Jyers/Marlin/discussions/1787#discussioncomment-1978481).

**Reply by TekoXVI** [reply]
Thanks so much for the quick reply!

**Reply by BlueWings172** [reply]
most welcome. Hope it gets fixed.

**Reply by TekoXVI** [reply]
Sorry I thought I had this figured out, but I'm on linux and trying to compile this after editing. Do you have any advice on how to do that?
edit: I didn't realize vs code was available for linux. I thought it was a windows thing. oops.

**Reply by BlueWings172** [reply]
Great hope it worked for you.
